src/Repositories/RedisJobRepository.php

```diff
@@ -688,6 +688,27 @@ public function deleteFailed($id)
         $this->connection()->del($id);
     }
 
+    /**
+     * Delete pending and reserved jobs for a queue.
+     *
+     * @param  string  $queue
+     * @return void
+     */
+    public function purge($queue)
+    {
+        $ids = collect(range(0, ceil($this->nextJobId() / 50)))->transform(function ($_, $index) use ($queue) {
+            return $this->getRecent($index === 0 ? null : $index * 50)->filter(function ($recent) use ($queue) {
+                return in_array($recent->status, ['pending', 'reserved']) && $recent->queue === $queue;
+            });
+        })->flatten(1)->pluck('id')->all();
```

Comment:
Please explain what this is doing.

Reply:
Here we're collecting the IDs of the pending or reserved jobs on the specified queue. To do so, we get all the recent jobs (50 at a time based on the in-built RedisJobRepository methods) and filter through them to check 1) if they're either pending or reserved and 2) if the queue matches the specified queue. On the filtered set of jobs, we pluck the job IDs and then delete them from the Redis hash and recent/pending jobs sorted sets.